app/controllers/signup.js

```diff
@@ -1,194 +1,219 @@
-    const index = this.fields.findIndex(field => field.id === name)
-    
-    set(this.formData, name, value)
-    
+    const index = this.fields.findIndex((field) => field.id === name);
+    set(this.formData, name, value);
+
     if (this.fields[index].required && !value) {
-      this.isSubmitDisabled = true
-      return set(this.fields[index], 'showError', true)
+      this.isSubmitDisabled = true;
+      set(this.fields[index], "showError", true);
     } else if (!this.fields[index].validator) {
-      return set(this.fields[index], 'showError', false)
+      set(this.fields[index], "showError", false);
     }
 
-    const anyErrors = this.fields.reduce((prev, field) => {
-      if (prev) {
-        return prev
+    const anyErrors = this.fields.map((field) => {
+      if (field.required && this.formData[field.id] === "") {
+        return true;
       }
-      if (field.required && this.formData[field.id] === '') {
-        return true
-      }
-      return false
-    }, false)
+      return false;
+    });
 
-    if (anyErrors) {
-      this.isSubmitDisabled = true
+    if (anyErrors.filter(Boolean).length) {
+      this.isSubmitDisabled = true;
+    } else {
+      this.isSubmitDisabled = false;
     }
   }
 
-  @action phoneNumberValidator(phoneNumber) {
-    if (typeof phoneNumber !== 'string'){ return false }
+  @action phoneNumberValidator(phone_number) {
+    if (typeof phone_number !== "string") {
+      return false;
+    }
 
-    const pattern=/^(0|[+91]{3})?[7-9][0-9]{9}$/;
-    const index = this.fields.findIndex((field) => field.id === 'phoneNumber')
+    const pattern = /^(0|[+91]{3})?[7-9][0-9]{9}$/;
+    const index = this.fields.findIndex((field) => field.id === "phone_number");
 
-    if (pattern.test(phoneNumber)) {
-      set(this.fields[index], 'showError', false);
+    if (pattern.test(phone_number)) {
+      set(this.fields[index], "showError", false);
     } else {
-      set(this.fields[index], 'showError', true);
+      set(this.fields[index], "showError", true);
     }
   }
 
   @action emailValidator(email) {
-    if (typeof email !== 'string') return false;
-    
-    const pattern = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
-    const index = this.fields.findIndex((field) => field.type === 'email')
+    if (typeof email !== "string") return false;
+
+    const pattern = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+    const index = this.fields.findIndex((field) => field.type === "email");
 
     if (pattern.test(email)) {
-      set(this.fields[index], 'showError', false);
+      set(this.fields[index], "showError", false);
     } else {
-      set(this.fields[index], 'showError', true);
+      set(this.fields[index], "showError", true);
     }
+    return;
   }
 
-  @action handleSubmit() {
+  @action async handleSubmit(e) {
     // submit
+    // https://github.com/Real-Dev-Squad/website-api-contracts/tree/main/users#patch-usersself
```

Comment:
Do we need this comment here?

Reply:
Just to remember the contract while working on this piece of code.